tools/FakeItEasy.Build/Program.cs

```diff
@@ -98,11 +99,12 @@ public static void Main(string[] args)
                     DependsOn("initialize-user-properties"),
                     () =>
                         {
-                            var originalText = File.ReadAllText("FakeItEasy.user.props", Encoding.UTF8);
-                            var newText = Regex.Replace(originalText, "(?<=<BuildProfile>)[^>]*(?=</BuildProfile>)", profile);
-                            if (newText != originalText)
+                            var xmlDoc = XDocument.Load("FakeItEasy.user.props", LoadOptions.PreserveWhitespace);
+                            var buildProfileElement = xmlDoc.Descendants("BuildProfile").FirstOrDefault();

```

Comment:
I tend to avoid using `Descendants`, because it can match elements unrelated to the one we're looking for (e.g. a `BuildProfile` element in an `ItemGroup`). I usually try to specify the path more precisely, e.g. `xmlDoc.Root.Elements("PropertyGroup").Elements("BuildProfile").FirstOrDefault()`.

Anyway, the current code doesn't handler the case where the `PropertyGroup` or `BuildProfile` element is missing... You probably need something like this:

```csharp
var buildProfileElement = xmlDoc.Root.Elements("PropertyGroup").Elements("BuildProfile").FirstOrDefault();
if (buildProfileElement is null)
{
    var propertyGroupElement = xmlDoc.Root.Element("PropertyGroup");
    if (propertyGroupElement is null)
    {
        propertyGroupElement = new XElement("PropertyGroup");
        xmlDoc.Root.Add(propertyGroupElement);
    }
    buildProfileElement = new XElement("BuildProfile");
    propertyGroupElement.Add(buildProfileElement);
}
...
```

Well, it ended up being much longer than the original version... But to be fair, it also handles more cases!

Reply:
I figured there'd be a low chance that the file would exist and be that malformed. However, I wasn't considering that users would fail to be notified that the transformation wasn't made. Glad you thought of it.

Pushed a fixup. I deviated from your suggestions slightly

1. I started checking for missing elements at PropertyGroup, not BuildProfile
2. I assumed only a singled BuildProfile element in the PropertyGroup
3. (more a deviation from my original), I stopped preserving whitespace on load, since it easily allowed me to indent the output on save. Without that change, newly-added elements were on the same line as their parents

Reply:
… and added a new fixup, since you seem to not have digested the last one yet. I extracted `GetOrAddElement`.